src/Compilers/CSharp/Test/Semantic/Semantics/StaticNullChecking.cs

```diff
@@ -4316,22 +4597,66 @@ public void RefOutParameters_05()
-        G(out x, ref y);
+        G(out x, ref y, in z);
+        x.ToString();
+        y.ToString();
+        z.ToString();
     }
-    static void G(out object x, ref object y)
+    static void G(out object x, ref object y, in object z)
     {
         x = new object();
     }
 }";
             var comp = CreateStandardCompilation(source, parseOptions: TestOptions.Regular8);
             comp.VerifyDiagnostics(
-                // (7,22): warning CS8604: Possible null reference argument for parameter 'y' in 'void C.G(out object x, ref object y)'.
-                //         G(out x, ref y);
-                Diagnostic(ErrorCode.WRN_NullReferenceArgument, "y").WithArguments("y", "void C.G(out object x, ref object y)").WithLocation(7, 22));
+                // (5,22): warning CS8604: Possible null reference argument for parameter 'y' in 'void C.G(out object x, ref object y, in object z)'.
+                //         G(out x, ref y, in z);
+                Diagnostic(ErrorCode.WRN_NullReferenceArgument, "y").WithArguments("y", "void C.G(out object x, ref object y, in object z)").WithLocation(5, 22),
+                // (5,28): warning CS8604: Possible null reference argument for parameter 'z' in 'void C.G(out object x, ref object y, in object z)'.
+                //         G(out x, ref y, in z);
+                Diagnostic(ErrorCode.WRN_NullReferenceArgument, "z").WithArguments("z", "void C.G(out object x, ref object y, in object z)").WithLocation(5, 28),
+                // (8,9): warning CS8602: Possible dereference of a null reference.
+                //         z.ToString();
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "z").WithLocation(8, 9));
+        }
+
+        [Fact]
+        public void RefOutParameters_06()
+        {
+            var source =
+@"class C
+{
+    static void F(object x, object y, object z)
+    {
+        G(out x, ref y, in z);
+        x.ToString();
+        y.ToString();
+        z.ToString();
+    }
+    static void G(out object? x, ref object? y, in object? z)
+    {
+        x = new object();
+    }
+}";
+            var comp = CreateStandardCompilation(source, parseOptions: TestOptions.Regular8);
+            comp.VerifyDiagnostics(
+                // (5,15): warning CS8600: Converting null literal or possible null value to non-nullable type.
+                //         G(out x, ref y, in z);
+                Diagnostic(ErrorCode.WRN_ConvertingNullableToNonNullable, "x").WithLocation(5, 15),
+                // (5,22): warning CS8600: Converting null literal or possible null value to non-nullable type.
+                //         G(out x, ref y, in z);
+                Diagnostic(ErrorCode.WRN_ConvertingNullableToNonNullable, "y").WithLocation(5, 22),
+                // (5,28): warning CS8620: Nullability of reference types in argument of type 'object' doesn't match target type 'object' for parameter 'z' in 'void C.G(out object? x, ref object? y, in object? z)'.
```

Comment:
>// (5,28): warning CS8620: Nullability of reference types in argument of type 'object' doesn't match target type 'object' for parameter 'z' in 'void C.G(out object? x, ref object? y, in object? z)'. [](start = 16, length = 198)

We probably should not report this warning for ```in``` parameter. It is fine to pass not null value in place of a nullable one. I think this should be similar to the way ByVal parameters a handled, since this is ```in``` the value is not going to be changed on the way out.